eng/common/scripts/Delete-RemoteBranches.ps1

```diff
@@ -45,17 +47,28 @@ foreach ($branch in $branches)
   $openPullRequests = $pullRequests | ? { $_.State -eq "open" }
   if ($openPullRequests.Count -gt 0)
   {
-    LogDebug "Branch [ $branchName ] in repo [ $RepoName ] has open pull Requests. Skipping"
+    LogDebug "Branch [ $branchName ] in repo [ $RepoId ] has open pull Requests. Skipping"
     LogDebug $openPullRequests.url
     continue
   }
 
-  LogDebug "Branch [ $branchName ] in repo [ $RepoName ] has no associated open Pull Request. Deleting Branch"
-  try{
-    Remove-GitHubSourceReferences -RepoOwner $RepoOwner -RepoName $RepoName -Ref ($branch.Remove(0,5)) -AuthToken $AuthToken
+  if ($LastCommitOlderThan) {
+    if (!$res.object -or !$res.object.url) {
+      LogWarning "No commit url returned from response. Skipping... "
+      continue
+    }
+    $commitDate = Get-GithubReferenceCommitDate -commitUrl $res.object.url
+    if ($commitDate -and ($commitDate -gt $LastCommitOlderThan)) {
+      LogDebug "The branch $branch last commit date $commitDate is newer than the date $LastCommitOlderThan. Skipping."
+      continue
+    }
+  } 
+  LogDebug "Branch [ $branchName ] in repo [ $RepoId ] has no associated open Pull Request. "
+  try {
+    Remove-GitHubSourceReferences -RepoId $RepoId -Ref $branch -AuthToken $AuthToken
```

Comment:
Should this be $branchName?

Reply:
I thought this function didn't like names like "refs/*"

Reply:
Not intend to make this change. It is supposed to be `$branch -replace "refs/"`